CMakeLists.txt

```diff
@@ -1,32 +1,60 @@
 cmake_minimum_required(VERSION 3.22.0)
-set(CMAKE_CXX_COMPILER "/usr/bin/clang++-18")
-
-project(iris LANGUAGES CXX)
-option(RUN_TESTS "Run unit tests" OFF)
-option(ENABLE_COVERAGE "Generate coverage report" OFF)
-option(ENABLE_ASAN "Compile with AddressSanitizer" OFF)
-option(RELEASE "Release build" OFF)
 
 set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_EXTENSIONS ON)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_COLOR_DIAGNOSTICS ON)
+set(CMAKE_DIRECTORY_LABELS ON)
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
-set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 
-if(RELEASE)
-    set(CMAKE_BUILD_TYPE Release)
-else()
-    set(CMAKE_BUILD_TYPE Debug)
-endif()
+project(iris
+  LANGUAGES
+    C CXX
```

Comment:
Why do we need to specify C here as well?

Reply:
that just came with some other stuff I copied out of another one of my repos. A bunch of stuff out of here should start to look pretty familiar haha... I use this repo as a base template for most of my c++ projects: https://github.com/vorlac/cmake-vcpkg-project-template/blob/main/CMakeLists.txt#L17-L22